### PR TITLE
Update embedded-hal

### PR DESCRIPTION
**Pull request by LehMaxence**
Removed write-only and read-only spi implementations
Handled Delay operation between transfers

### CODE REVIEW

**Comment by LehMaxence** [comment]
I tested this with hardware today on a Raspberry Pi 4 and a ADXL355 accelerometer connected over SPI with this code:
```rust
    let mut spi_dev = Spidev::open(SPI_PATH)?;
    let options = SpidevOptions::new()
        .bits_per_word(8)
        .max_speed_hz(10_000_000)
        .mode(SpiModeFlags::SPI_MODE_0)
        .build();
    spi_dev.configure(&options)?;

    let mut buf_1 = [0u8; 1];
    let mut buf_2 = [0u8; 1];

    let mut operations = [
        SpiOperation::DelayUs(1_000_000),
        SpiOperation::Write(&[(0x2D << 1), 0x01]),
        SpiOperation::DelayUs(500),
        SpiOperation::Write(&[(0x2D << 1) | 0x01]),
        SpiOperation::Read(&mut buf_1),
        SpiOperation::DelayUs(500),
        SpiOperation::Write(&[(0x2D << 1), 0x00]),
        SpiOperation::DelayUs(500),
        SpiOperation::Write(&[(0x2D << 1) | 0x01]),
        SpiOperation::Read(&mut buf_2),
        SpiOperation::DelayUs(5_000_000),
    ];

    spi_dev.transaction(&mut operations)?;

    println!("buf_1: {:X}", buf_1[0]);
    println!("buf_2: {:X}", buf_2[0]);
```

It worked as expected.

**Review by eldruin — COMMENTED** [review]
Thank you for your work here.
Unfortunately, this approach is suboptimal in that it requires multiple transfers and flushes on the bus every time there is a delay, even though the underlying linux device supports delay transfers.

The optimal approach here would be to first extend spidev: https://github.com/rust-embedded/rust-spidev/pull/40 to support delay operations and then updating this crate in line with what was outlined in: https://github.com/rust-embedded/linux-embedded-hal/issues/87#issuecomment-1267641344

Sorry for the confusion and lack of more explicit guidance.

File: src/spi.rs [hunk 1]
```diff
@@ -132,6 +96,9 @@ mod embedded_hal_impl {
                         };
                         SpidevTransfer::read_write(tx, buf)
                     }
+                    SpiOperation::DelayUs(us) => {
+                        SpidevTransfer::delay((*us).try_into().unwrap_or(u16::MAX))
```
Comment (LehMaxence) [inline src/spi.rs:100]: I chose to max out the delay to `u16::MAX` but that may not be what we want here. 
Let me know what you think.

Comment (eldruin) [reply src/spi.rs:100]: I think we can leave at that here but please document in this method and in the `Spidev` struct that delays are capped to 65535 microseconds.

**Review by eldruin — COMMENTED** [review]
Thank you!

**Review by eldruin — APPROVED** [review]
Sorry for the delay.
Looks good to me, thank you!

**Comment by eldruin** [comment]
Ach, CI has to be migrated to GitHub merge queue for this to be merged.

**Comment by eldruin** [comment]
@LehMaxence could you rebase this to master now that CI is fixed?

**Comment by LehMaxence** [comment]
> @LehMaxence could you rebase this to master now that CI is fixed?

Rebased :-)

**Review by eldruin — APPROVED** [review]
Great, thank you!

**Pull request merged**